public/app/plugins/datasource/cloudwatch/datasource.ts

```diff
@@ -504,6 +505,14 @@ export class CloudWatchDatasource
 
         const lastError = findLast(res.results, (v) => !!v.error);
 
+        dataframes.forEach((frame) => {
+          frame.fields.forEach((field) => {
+            if (field.type === FieldType.time) {
+              field.config.interval = frame.meta?.custom?.period * 1000;
+            }
+          });
+        });
```

Comment:
Adding config.interval to every dataframe field = that will allow the frontend to show the gap 👍

Reply:
can you add a comment here saying that so we know for the future?